File: services/auth-service/src/main/java/com/otterworks/auth/exception/ApiErrorResponse.java
package com.otterworks.auth.exception;

public record ApiErrorResponse(ApiError error) {

  public record ApiError(String code, String message, int status) {}

  public static ApiErrorResponse of(String code, String message, int status) {
    return new ApiErrorResponse(new ApiError(code, message, status));
  }
}

File: services/audit-service/src/Models/ApiErrorResponse.cs
namespace OtterWorks.AuditService.Models;

public sealed record ApiError(string Code, string Message, int Status);

public sealed record ApiErrorResponse(ApiError Error)
{
    public static ApiErrorResponse Create(string code, string message, int status)
    {
        return new ApiErrorResponse(new ApiError(code, message, status));
    }
}
